mllib/src/main/scala/org/apache/spark/ml/functions.scala

```diff
@@ -39,10 +39,31 @@ object functions {
     }
   }.asNonNullable()
 
+  private val vectorToArrayFloatUdf = udf { vec: Any =>
+    vec match {
+      case v: Vector => v.toArray.map(_.toFloat)
+      case v: OldVector => v.toArray.map(_.toFloat)
```

Comment:
Why not merge this method into `vectorToArrayUdf` ? like:
`vectorToArrayUdf = udf { (vec: Any, type: String) => ... }`

Reply:
@liangz1 It's my fault. We cannot combine the two UDF together. Their return type are Array<Double> and Array<Float>, they're incompatible. This leads to error in your tests.